### PR TITLE
Image cropping

### PR DESCRIPTION
**Pull request by airmiha**
### Description (what's changed?)

Ability to crop image for image to image by pan and zoom.

### CODE REVIEW

**Comment by kvarela** [comment]
Task linked: [CU-86aykjmze Implement image cropping tool for image to image](https://app.clickup.com/t/86aykjmze)

**Comment by kvarela** [comment]
Task linked: [CU-86aykjmze Implement image cropping tool for image to image](https://app.clickup.com/t/86aykjmze)

File: src/lib/editor/toolbar/image-to-image/cropImage.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+export const createImage = (url: string): Promise<HTMLImageElement> =>
+  new Promise((resolve, reject) => {
+    const image = new Image();
+    image.addEventListener('load', () => resolve(image));
+    image.addEventListener('error', (error) => reject(error));
+    image.setAttribute('crossOrigin', 'anonymous'); // needed to avoid cross-origin issues on CodeSandbox
```
Comment (kvarela) [inline src/lib/editor/toolbar/image-to-image/cropImage.ts:6]: Is this still necessary?

File: src/lib/editor/toolbar/image-to-image/index.tsx [hunk 2]
```diff
@@ -90,10 +105,19 @@ export default function ImageToImageGenerator({
...
 
-    setUploadedImage(null);
+      setSelectedInputImage({
+        preview: URL.createObjectURL(croppedImage),
+        file: new File([croppedImage], 'image', { type: 'image/png' }),
+      });
+
+      setUploadedImage(null);
+    } catch (e) {
+      console.error(e);
```
Comment (kvarela) [inline src/lib/editor/toolbar/image-to-image/index.tsx:119]: Should probably show an error toast or something if this doesn't work

**Review by kvarela — CHANGES_REQUESTED** [review]

**Review by btosic — APPROVED** [review]

**Review by kvarela — APPROVED** [review]

**Pull request merged**